### PR TITLE
#515 aspects ordering update

### PR DESCRIPTION
**Pull request by Romeh**
spring aspect ordering enabling except bulkhead

### CODE REVIEW

File: resilience4j-spring/src/main/java/io/github/resilience4j/circuitbreaker/configure/CircuitBreakerConfigurationProperties.java [hunk 1]
```diff
@@ -21,14 +21,22 @@
...
 public class CircuitBreakerConfigurationProperties extends io.github.resilience4j.common.circuitbreaker.configuration.CircuitBreakerConfigurationProperties {
-
+	private int circuitBreakerAspectOrder = Ordered.LOWEST_PRECEDENCE - 2;
 	/**
-	 * As of release 0.16.0 as we set an implicit spring aspect order now which is retry then circuit breaker then rate limiter then bulkhead
+	 * As of release 0.16.0 as we set an implicit spring aspect order now which is retry then circuit breaker then rate limiter then bulkhead but
+	 * user can override it still if he has different use case but bulkhead will be first aspect all the time due to the implicit order we have it for bulkhead
 	 */
 	public int getCircuitBreakerAspectOrder() {
 		return Ordered.LOWEST_PRECEDENCE - 2;
```
Comment (RobWin) [inline resilience4j-spring/src/main/java/io/github/resilience4j/circuitbreaker/configure/CircuitBreakerConfigurationProperties.java]: This should read the variable.

File: resilience4j-spring/src/main/java/io/github/resilience4j/ratelimiter/configure/RateLimiterConfigurationProperties.java [hunk 2]
```diff
@@ -18,11 +18,21 @@
...
 public class RateLimiterConfigurationProperties extends io.github.resilience4j.common.ratelimiter.configuration.RateLimiterConfigurationProperties {
-
+	private int rateLimiterAspectOrder = Ordered.LOWEST_PRECEDENCE - 1;
 	/**
-	 * As of release 0.16.0 as we set an implicit spring aspect order now which is retry then circuit breaker then rate limiter then bulkhead
+	 * As of release 0.16.0 as we set an implicit spring aspect order now which is retry then circuit breaker then rate limiter then bulkhead but
+	 * user can override it still if he has different use case but bulkhead will be first aspect all the time due to the implicit order we have it for bulkhead
 	 */
 	public int getRateLimiterAspectOrder() {
 		return Ordered.LOWEST_PRECEDENCE - 1;
```
Comment (RobWin) [inline resilience4j-spring/src/main/java/io/github/resilience4j/ratelimiter/configure/RateLimiterConfigurationProperties.java]: Variable

File: resilience4j-spring/src/main/java/io/github/resilience4j/retry/configure/RetryConfigurationProperties.java [hunk 3]
```diff
@@ -21,12 +21,22 @@
  * Main spring properties for retry configuration
  */
 public class RetryConfigurationProperties extends io.github.resilience4j.common.retry.configuration.RetryConfigurationProperties {
-
+	private int retryAspectOrder = Ordered.LOWEST_PRECEDENCE - 3;
 	/**
 	 * As of release 0.16.0 as we set an implicit spring aspect order now which is retry then circuit breaker then rate limiter then bulkhead
+	 * but the user can override it still if he has different use case but bulkhead will be first aspect all the time due to the implicit order we have it for bulkhead
 	 */
 	public int getRetryAspectOrder() {
```
Comment (RobWin) [inline resilience4j-spring/src/main/java/io/github/resilience4j/retry/configure/RetryConfigurationProperties.java]: Variable

File: resilience4j-spring/src/main/java/io/github/resilience4j/circuitbreaker/configure/CircuitBreakerConfigurationProperties.java (hunk 1 above)
Comment (Romeh) [reply resilience4j-spring/src/main/java/io/github/resilience4j/circuitbreaker/configure/CircuitBreakerConfigurationProperties.java]: rushing mistakes :) , updated

**Comment by Romeh** [comment]
> I think we should add a test

Done and added plus spring boot tests also test the setting of the property as i added them back into the application yml files

**Pull request merged**